apps/prairielearn/src/lib/groups.ts

```diff
@@ -47,97 +50,83 @@ const GroupRoleSchema = z.object({
 });
+type GroupRole = z.infer<typeof GroupRoleSchema>;
 
 const AssessmentLevelPermissionsSchema = z.object({
   can_assign_roles_at_start: z.boolean().nullable(),
   can_assign_roles_during_assessment: z.boolean().nullable(),
 });
+type AssessmentLevelPermissions = z.infer<typeof AssessmentLevelPermissionsSchema>;
+
+interface RolesInfo {
+  roleAssignments: Record<string, RoleAssignment[]>;
+  groupRoles: GroupRole[];
+  validationErrors: GroupRole[];
+  disabledRoles: string[];
+  rolesAreBalanced: boolean;
+  usersWithoutRoles: GroupMember[];
+}
 
-/** @typedef {z.infer<GroupConfigSchema>} GroupConfig */
-/** @typedef {z.infer<GroupMemberSchema>} GroupMember */
-/** @typedef {z.infer<GroupRoleSchema>} GroupRole */
-/** @typedef {z.infer<AssessmentLevelPermissionsSchema>} AssessmentLevelPermissions */
-
-/**
- * @typedef {Object} RolesInfo
- * @property {Object} roleAssignments
- * @property {GroupRole[]} groupRoles
- * @property {GroupRole[]} validationErrors
- * @property {string[]} disabledRoles
- * @property {boolean} rolesAreBalanced
- * @property {GroupMember[]} usersWithoutRoles
- */
-
-/**
- * @typedef {Object} GroupInfo
- * @property {GroupMember[]} groupMembers
- * @property {number} groupSize
- * @property {string} groupName
- * @property {string} joinCode
- * @property {boolean} start
- * @property {RolesInfo} rolesInfo
- */
+interface GroupInfo {
+  groupMembers: GroupMember[];
+  groupSize: number;
+  groupName: string;
+  joinCode: string;
+  start: boolean;
+  rolesInfo?: RolesInfo;
+}
 
-/**
- * @typedef {Object} GroupRoleAssignment
- * @property {string} group_role_id
- * @property {string} user_id
- */
+interface GroupRoleAssignment {
+  group_role_id: string;
+  user_id: string;
+}
 
 /**
  * Gets the group config info for a given assessment id.
- *
- * @param {string} assessmentId
- * @returns {Promise<GroupConfig>}
  */
-export async function getGroupConfig(assessmentId) {
-  let params = { assessment_id: assessmentId };
-  return await sqldb.queryValidatedOneRow(sql.get_group_config, params, GroupConfigSchema);
+export async function getGroupConfig(assessmentId: string): Promise<GroupConfig> {
+  return await sqldb.queryRow(
+    sql.get_group_config,
+    { assessment_id: assessmentId },
+    GroupConfigSchema,
+  );
 }
 
 /**
  * Returns the group id for the user's current group in an assessment, if it exists.
  * Used in checking whether the user is in a group or not.
- *
- * @param {string} assessmentId
- * @param {string} userId
- * @returns {Promise<string | null>}
  */
-export async function getGroupId(assessmentId, userId) {
-  let params = { assessment_id: assessmentId, user_id: userId };
-  return await sqldb.queryValidatedSingleColumnZeroOrOneRow(
+export async function getGroupId(assessmentId: string, userId: string): Promise<string | null> {
+  return await sqldb.queryOptionalRow(
     sql.get_group_id,
-    params,
+    { assessment_id: assessmentId, user_id: userId },
     GroupIdSchema,
```

Comment:
Nit: I don't think this needs a dedicated type, consider using `IdSchema` instead (I know you didn't change this, but as long as we're touching this file...)

Reply:
Done in 38cdf5a.